### PR TITLE
Important fixes to build_materialx_osl.py

### PR DESCRIPTION
**Pull request by amartinez-cg**
previously script was overwriting .osl files

added some checking to make sure source and destination exist

corrected osl file counter

commented functions

## Checklist:

- [ ] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [ ] I have previously submitted a [Contributor License Agreement](http://opensource.imageworks.com/cla/).
- [ ] I have updated the documentation, if applicable.
- [ ] I have ensured that the change is tested somewhere in the testsuite (adding new test cases if necessary).
- [ ] My code follows the prevailing code style of this project.

### CODE REVIEW

File: src/build-scripts/build_materialX_osl.py [hunk 1]
```diff
@@ -15,14 +15,16 @@
...
 
+# global defaults
 MX_SOURCE = '../shaders/MaterialX'
 OSL_DEST = '../../dist/linux64/shaders/MaterialX'
 
 MX_SOURCE_EXT = 'mx'
 
 OSLC_CMD = 'oslc '
-OSLC_PATH = ''
+OSLC_PATH = '../../dist/linux64/bin/oslc'
```
Comment (lgritz) [inline src/build-scripts/build_materialX_osl.py]: You can't have this hard-coded to the architecture

**Comment by amartinez-cg** [comment]
Not sure what else to do about the architecture path that provide a global variable that can be changed by hand.

File: src/build-scripts/build_materialX_osl.py [hunk 2]
```diff
@@ -14,15 +14,19 @@
 import os
 import sys
 import re
+import argparse
 
+# global defaults
+ARCH = 'linux64'
 MX_SOURCE = '../shaders/MaterialX'
-OSL_DEST = '../../dist/linux64/shaders/MaterialX'
+OSL_DEST = '../../dist/%s/shaders/MaterialX'%ARCH
```
Comment (lgritz) [inline src/build-scripts/build_materialX_osl.py]: This should probably be 'build', not 'dest'.
The 'make install' should copy things from build to dest, but none of the build process should put things directly in dest.

**Comment by amartinez-cg** [comment]
10-4!

File: src/build-scripts/build_materialX_osl.py [hunk 3]
```diff
@@ -14,15 +14,19 @@
 import os
 import sys
 import re
+import argparse
 
+# global defaults
+ARCH = 'linux64'
```
Comment (lgritz) [inline src/build-scripts/build_materialX_osl.py]: I'm thinking that this python script should take the architecture (or, even better, just the paths to oslc and the shaders/MaterialX directory), which CMakeLists.txt can easily pass to it.

If the oslc path is not supplied, it should just assume 'oslc' is in the execution path, and if the build/shaders area isn't supplied, it should just assume the current dir.

Comment (lgritz) [reply src/build-scripts/build_materialX_osl.py]: I mean "take" -> as parameters to this script

**Comment by amartinez-cg** [comment]
This latest commit does that, but keeps the global declarations, which is
confusing.  I think Ill get rid of the globals and pass the arguments
through the functions.

**Pull request merged**